src/librustc_mir_build/hair/pattern/check_match.rs

```diff
@@ -596,11 +601,11 @@ fn check_legality_of_move_bindings(cx: &mut MatchVisitor<'_, '_>, has_guard: boo
 
     // Found some bad by-move spans, error!
     if !by_move_spans.is_empty() {
-        let mut err = struct_span_err!(
-            sess,
-            MultiSpan::from_spans(by_move_spans.clone()),
-            E0009,
-            "cannot bind by-move and by-ref in the same pattern",
+        let mut err = feature_err(
+            &sess.parse_sess,
+            sym::move_ref_pattern,
+            by_move_spans.clone(),
+            "binding by-move and by-ref in the same pattern is unstable",
```

Comment:
Any reason _not_ to keep the error code?

Reply:
Doing `.code(error_code!(...))` would override the existing error code for unstable features, so useful information is removed that way. Directing the user towards the old error code with an explanation that this is a temporary situation is feels rather roundabout as most of the historical information is irrelevant anyways. I think this is more direct this way (and it reduces the amount of work when stabilizing).